pkg/controllers/updaterun/execution.go

```diff
@@ -172,17 +177,27 @@ func (r *Reconciler) executeUpdatingStage(
 			markClusterUpdatingFailed(clusterStatus, updateRun.Generation, unexpectedErr.Error())
 			return 0, fmt.Errorf("%w: %s", errStagedUpdatedAborted, unexpectedErr.Error())
 		}
+
 		finished, updateErr := checkClusterUpdateResult(binding, clusterStatus, updatingStageStatus, updateRun)
 		if finished {
 			finishedClusterCount++
+			unmarkUpdateRunStuck(updateRun)
```

Comment:
why not setting the progressing as false and saying that it's completed in the reason?

removing the condition looks weird here.

Reply:
Only sinlge cluster completes, not the updateRun. The updateRun should still be in Progressing status. Here if the cluster update completes, we are sure that it's not stuck, and we remove that.